sql/core/src/main/scala/org/apache/spark/sql/SparkSession.scala

```diff
@@ -279,6 +281,27 @@ class SparkSession private(
       initialSessionOptions)
   }
 
+  /**
+   * Clone JobArtifact set info from parent session while call cloneSession api.
+   *  
+   * @since 3.5.1 
+   */     
+  private[sql] def cloneJobArtifactSet(parentSessionId: String,
+                                       childSessionId: String): Unit = {
+      sparkContext.addedJars.get(parentSessionId).map{ jars =>
+      val newJars = new ConcurrentHashMap[String, Long]().asScala ++= jars
+      sparkContext.addedJars.put(childSessionId, newJars)
+    }
+    sparkContext.addedFiles.get(parentSessionId).map{ files =>
+      val newFiles = new ConcurrentHashMap[String, Long]().asScala ++= files
+      sparkContext.addedFiles.put(childSessionId, newFiles)
+    }
+    sparkContext.addedArchives.get(parentSessionId).map{ archives =>
```

Comment:
```suggestion
    sparkContext.addedArchives.get(parentSessionId).foreach { archives =>
```

Reply:
ok, i will optimize the code by your advice